src/sentry/workflow_engine/endpoints/organization_available_action_index.py

```diff
@@ -0,0 +1,138 @@
+            ),
+            400: RESPONSE_BAD_REQUEST,
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def get(self, request, organization):
+        """
+        Returns a list of available actions for a given org
+        """
+        provider_integrations: dict[str, list[RpcIntegration]] = defaultdict(list)
+        for integration in get_available_action_integrations_for_org(organization):
+            provider_integrations[integration.provider].append(integration)
+
+        sentry_app_component_contexts = app_service.get_installation_component_contexts(
+            filter={"organization_id": organization.id},
+            component_type="alert-rule-action",
+            include_contexts_without_component=True,
+        )
+
+        actions = []
+        for action_type, handler in action_handler_registry.registrations.items():
+            # add integration actions
+            if hasattr(handler, "provider_slug"):
+                integrations = provider_integrations.get(handler.provider_slug, [])
+                if integrations:
+                    actions.append(
+                        serialize(
+                            handler,
+                            request.user,
+                            ActionHandlerSerializer(),
+                            action_type=action_type,
+                            integrations=integrations,
+                        )
+                    )
+
+            # add alertable sentry app actions
+            elif action_type == Action.Type.SENTRY_APP:
+                for context in sentry_app_component_contexts:
+                    if context.installation.sentry_app.is_alertable:
+                        actions.append(
+                            serialize(
+                                handler,
+                                request.user,
+                                ActionHandlerSerializer(),
+                                action_type=action_type,
+                                sentry_app_context=context,
+                            )
+                        )
+
+            # add plugin service actions
+            elif action_type == Action.Type.WEBHOOK:
+                plugins = get_notification_plugins_for_org(organization)
+                if plugins:
+                    actions.append(
+                        serialize(
+                            handler,
+                            request.user,
+                            ActionHandlerSerializer(),
+                            action_type=action_type,
+                            services=plugins,
+                        )
+                    )
+
+            # add all other actions
+            else:
+                actions.append(
+                    serialize(
+                        handler, request.user, ActionHandlerSerializer(), action_type=action_type
+                    )
+                )
```

Comment:
What types of actions are not integrations, sentry apps, or plugins? Webhooks?

Reply:
the only actions that hit this else case are email and "plugin", which actually represents the action for sending a notification to all legacy integrations

Reply:
maybe stupid question but then what's the difference between this and the plugins in the elif above?

Reply:
not stupid at all, the action types are very confusing. i'll list everything here:

1. `Action.Type.EMAIL` = email action
2. `Action.Type.SLACK`, `Action.Type.GITHUB`, etc = integration actions
3. `Action.Type.SENTRY_APP` = sentry app actions
4. `Action.Type.WEBHOOK` = action for sending a notification to a single legacy integration (here we list all plugins for the org for the user to select from)
5. `Action.Type.PLUGIN` = the action called "Send a notification to all legacy integrations" (this is handled in the `else` since we don't need to pass any extra information if we're sending to all integrations)